### PR TITLE
Soves several issues in com_content extension

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Solves https://github.com/OSTraining/OSMap/issues/100 (MAP-100)
Solves https://github.com/OSTraining/OSMap/issues/99 (MAP-99)

And makes a better relation with categories/articles links with parent menu item

I think now it has a better handle of categories and subcategories and their levels and their articles.

### CODE REVIEW

**Comment by nicksavov** [comment]
Hi André,

Thanks for the PR! I've assigned it to our developer, Anderson, for review.

**Comment by andergmartins** [comment]
Related to MAP-100 and MAP-99

**Comment by andrepereiradasilva** [comment]
this is broken ... again in current latest releases

**Comment by andergmartins** [comment]
@andrepereiradasilva Thanks, I will investigate

File: src/extensions/plg_osmap_com_content/com_content.php [hunk 1]
```diff
@@ -537,6 +547,13 @@ public static function includeCategoryContent($osmap, $parent, $catid, &$params,
                 $node->catslug = $item->catid;
                 $node->link    = ContentHelperRoute::getArticleRoute($node->slug, $node->catslug);
 
+                if (strpos($node->link,'Itemid=')===false) {
+                    $node->itemid = $itemid;
+                    $node->link .= '&Itemid='.$parent->id;
+                } else {
+                    $node->itemid = $itemid;
+					$node->link = preg_replace('/Itemid=([0-9]+)/','Itemid='.$parent->id,$node->link);
```
Comment (andergmartins) [inline src/extensions/plg_osmap_com_content/com_content.php]: We removed this since it was changing the SEF URLs for some links in the sitemap

**Comment by andergmartins** [comment]
Hi, I was double checking and there were many changes on the joomla plugin. But the unique thing I think is really different from the original PR is regarding the comment above

**Comment by andrepereiradasilva** [comment]
yups i now need to put the dupolicate results on! works that way! sorry, didn't notice this changed. Working now.

Anyway notice a bug: the url menu types does not appear in html sitemap now

**Comment by andergmartins** [comment]
Humm, The menu types, really? I probably missed that. Please, can you be more specific where you missed them?

**Comment by andrepereiradasilva** [comment]
I have a second level menu item with is url type.
That does not appear in my HTML sitemap.

**Comment by andergmartins** [comment]
Hi @andrepereiradasilva do you mean an External URL menu type?
Does it point to your own site (is an internal URL)?

**Comment by andrepereiradasilva** [comment]
again i'm sorry i noticed now there is a option now on the sitemap options to show or not the external url.

**Pull request merged**